File: 卡码/代码随想录/JS/动态规划/02-背包问题/02-完全背包/Kama_0057.js
//57. 爬楼梯
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

(async function(){
	const [n,m]=(await readline()).split(" ").map(Number);
	const dp=new Array(n+1).fill(0);
	dp[0]=1;
	for(let i=1;i<=n;i++){
		for(let j=1;j<=m;j++){
			if(i>=j) dp[i]+=dp[i-j];
			else break;
		}
	}
	console.log(dp[n]);
})();

File: 卡码/代码随想录/Python/动态规划/02-背包问题/02-完全背包/Kama_0057.py
#57. 爬楼梯
from typing import *
import sys

def main():
  n,m=map(int,input().split())
  dp:List[int]=[0]*(n+1)
  dp[0]=1
  for i in range(1,n+1):
    for j in range(1,m+1):
      if i>=j:
        dp[i]+=dp[i-j]
      else:
        break
  print(dp[n])

if __name__=="__main__":
  main()